eng/common/scripts/Detect-Api-Changes.ps1

```diff
@@ -0,0 +1,98 @@
+  [Parameter(Mandatory=$True)]
+  [string] $CommitSha,
+  [Parameter(Mandatory=$True)]
+  [string] $ArtifactList,
+  [string] $ArtifactName = "packages",
+  [string] $APIViewUri = "https://apiview.dev/PullRequest/DetectApiChanges"
+)
+
+# Submit API review request and return status whether current revision is approved or pending or failed to create review
+function Submit-Request($filePath)
+{
+    $repoName = "azure-sdk-for-$Language"
+    $queryParam = "artifactName=$ArtifactName&buildId=$BuildId&filePath=$filePath&commitSha=$CommitSha&repoName=$repoName&pullRequestNumber=$PullRequestNumber"    
+    $uri= "$($APIViewUri)?$($queryParam)"
+    Write-Host "Request URI: $uri"
+    try
+    {
+        $Response = Invoke-WebRequest -Method 'GET' -Uri $uri
+        $StatusCode = $Response.StatusCode
+    }
+    catch
+    {
+        Write-Host "Exception details: $($_.Exception.Response)"
+        $StatusCode = $_.Exception.Response.StatusCode
+    }
+
+    return $StatusCode
+}
+
+function Shoud-Process-Package($pkgPath, $packageName)
+{
+    $pkg = Split-Path -Leaf $pkgPath
+    $configFileDir = Join-Path -Path $ArtifactPath "PackageInfo"
+    $pkgPropPath = Join-Path -Path $configFileDir "$packageName.json"
+    if (-Not (Test-Path $pkgPropPath))
+    {
+        Write-Host " Package property file path $($pkgPropPath) is invalid."
+        return $False
+    }
+    # Get package info from json file created before updating version to daily dev
+    $pkgInfo = Get-Content $pkgPropPath | ConvertFrom-Json
+    Write-Host "SDK Type: $($pkgInfo.SdkType)"
+    return ($pkgInfo.SdkType -eq "client" -and $pkgInfo.IsNewSdk)
+}
+
+function Log-Input-Params()
+{
+    Write-Host "Artifact Path: $($ArtifactPath)"
+    Write-Host "Artifact Name: $($ArtifactName)"
+    Write-Host "PullRequest Number: $($PullRequestNumber)"
+    Write-Host "BuildId: $($BuildId)"
+    Write-Host "Language: $($Language)"
+    Write-Host "Commit SHA: $($CommitSha)"
+}
+
+. (Join-Path $PSScriptRoot common.ps1)
+Log-Input-Params
+
+$artifacts = $ArtifactList.Split(",")
+foreach ($pkgName in $artifacts)
+{
+    Write-Host "Processing $($pkgName)"
+    $packages = @{}
+    if ($FindArtifactForApiReviewFn -and (Test-Path "Function:$FindArtifactForApiReviewFn"))
+    {
+        $packages = &$FindArtifactForApiReviewFn $ArtifactPath $pkgName
+    }
+    else
+    {
+        Write-Host "The function for 'FindArtifactForApiReviewFn' was not found.`
+        Make sure it is present in eng/scripts/Language-Settings.ps1 and referenced in eng/common/scripts/common.ps1.`
+        See https://github.com/Azure/azure-sdk-tools/blob/main/doc/common/common_engsys.md#code-structure"
+        exit(1)
+    }
+
+
+    if ($packages)
+    {
+        $pkgPath = $packages.Values[0]
+        if (Shoud-Process-Package -pkgPath $pkgPath -packageName $pkgName)
+        {
+            $filePath = $pkgPath.Replace($ArtifactPath , "").Replace("\", "/")
+            Submit-Request -filePath $filePath
```

Comment:
Should we collect all status codes and fail the pipeline if any failed to submit?

Reply:
yes, I should actually make that change.

Reply:
Actually we do not fail the task even if it fails for a package so it will continue for all artifacts. I have verified this as part of testing against staging instance. API change detection step is only informing APIview to find any changes and comment on PR so this step in pipeline is more of a trigger to APIView.

Reply:
But in this case, isn't the potential failure that you failed to upload to APIView, in which case it wouldn't find any invalid changes? So you would want to alert here that not all changes have been analyzed.

Reply:
That's true. If any error occurs when sending trigger event, We need to show summary of failure. I will make a change to show a warning of failed packages. This log will be further enhanced once I have API View wiki updated with FAQ about change detection in PR pipeline.